### ISSUE TITLE
Can jrnl run without being installed?

### ISSUE THREAD

**Issue by CodyKochmann** [jrnl-org/jrnl#413]
I basically live off of the iPad in my free time and would love to make entries to jrnl from pythonista locally. The way it works is basically a directory where python can run so what you need to have is an executable script you can run that will start the python program without ever installing anything. 

This is also the ideal way to run python packages where you don't have permissions to install things such as a work machine without sudo access (where I would also like to put jrnl). 

Basically, what I'm wondering is if you know a way to run jrnl on a machine where the best that I can do is clone the directory and use the source as is without any further installation. I know it's possible and probably could cook together a standalone python script that mimics this program's behavior but, this package is already so solid and trustworthy that I would prefer to just run this. 

The execution would run through an alias somewhat like this `alias jrnl='~/jrnl/run.py'`

**Comment by MinchinWeb** [comment]
When installed, the console script is calling `jrnl.run()` (see [setup.py](https://github.com/maebert/jrnl/blob/master/setup.py#L156)). Will that be enough on your end?

Though not as convenient, when I'm on the go, I'll write notes as text files, upload them to dropbox, and then import them into jrnl (`jrnl < some_entry.txt`) when I'm back on the computer.

**Comment by philipsd6** [comment]
FWIW, I use [Workflow](https://workflow.is/) on my iPhone to update my jrnl via ssh. Basically looks like this:

1. Text: (Input)(Clipboard)(Ask When Run)
1. Set Variable: Entry
1. Run Script Over SSH:<br />
    `jrnl 2>&1 <<EOF`<br />
    (Input)<br />
    `EOF`
1. Replace Text:
    * Find Text=[\\[\\]]
    * Replace With=
    * Case Sensitive=True
    * Regular Expression=True
1. Show Alert:
    * Title=Jrnl
    * (Input)

Works well for me!

**Comment by MinchinWeb** [comment]
Would the solution to this be to make jrnl work from `python -m jrnl`?

**Comment by wren** [comment]
Hrmm, not sure. I think that, if we're going to try to get this running in a standalone fashion, I would tend towards a pex distribution rather than the `python -m` approach. My worry is that `python -m` has historically been a fairly difficult to do dependency management with.

That being said, I'm not sure how impactful this feature would be. This ticket doesn't seem to have had a lot of traffic over the years. Are there a lot of people interested in it?

**Comment by peter-schmidbauer** [comment]
They way poetry handles this is interesting, it basically vendors in all dependencies during the build process and therefore does not require pip or any external dependencies at all (only when installed using their [install script](https://github.com/sdispater/poetry/blob/master/get-poetry.py).

However, I don't believe all of this is really necessary. Don't install programs like jrnl [into your system python](https://www.reddit.com/r/Python/comments/9p2tkq/stop_using_sudo_pip_install/) (we shouldn't recommend doing so in the Readme either by the way), instead at least use `pip install --user`. Even better would be to use independent virtual environments to encapsulate dependencies from each other, tools like [pipx](https://github.com/pipxproject/pipx/) can do that for you which I find very handy.

**Comment by wren** [comment]
@peter-schmidbauer Agreed. I've been looking into tools like [pex](https://github.com/pantsbuild/pex) and pipx to move away from the current approach. In my experience, `pip` has been very unreliable, especially when dealing with cross-platform installs.

**Comment by wren** [comment]
I think we should probably recommend pipx for new installs. The only thing holding this up for me is updating the tests.
